src/csync/vio/csync_vio_local_win.cpp

```diff
@@ -41,81 +41,53 @@ Q_LOGGING_CATEGORY(lcCSyncVIOLocal, "sync.csync.vio_local", QtInfoMsg)
  * directory functions
  */
 
-typedef struct dhandle_s {
+struct csync_vio_handle_t {
   WIN32_FIND_DATA ffd;
   HANDLE hFind;
   int firstFind;
-  mbchar_t *path; // Always ends with '\'
-} dhandle_t;
+  QString path; // Always ends with '\'
+};
 
 static int _csync_vio_local_stat_mb(const mbchar_t *uri, csync_file_stat_t *buf);
 
-csync_vio_handle_t *csync_vio_local_opendir(const char *name) {
-  dhandle_t *handle = NULL;
-  mbchar_t *dirname = NULL;
+csync_vio_handle_t *csync_vio_local_opendir(const QString &name) {
 
-  handle = (dhandle_t*)c_malloc(sizeof(dhandle_t));
+    QScopedPointer<csync_vio_handle_t> handle(new csync_vio_handle_t{});
 
-  // the file wildcard has to be attached
-  int len_name = strlen(name);
-  if( len_name ) {
-      char *h = NULL;
+    // the file wildcard has to be attached
+    QString dirname = name + QLatin1String("/*");
```

Comment:
I don't see the `path_to_locale` call anywhere. It's the thing that also converts to UNC paths for long-filename compatibility.

Reply:
well spotted!